### ISSUE TITLE
[18.0.0.1] OpenTracing  NPE failure?

### ISSUE THREAD

**Issue by hrstoyanov** [OpenLiberty/open-liberty#3560]
I have a JAX-RS service that fails for some unrelated reason (which I have identified already). The problem is that OpenTracing in OL 18.0.0.1 fails too with NullPointerException, masking the real exception in my service. Can we make OpenTracing more robust and prevent it from obscuring the underlying cause? Here is the stacktrace:
```
Exception = java.lang.NullPointerException
Source = com.ibm.ws.opentracing.OpentracingUserFeatureAccessService
probeid = 70
Stack Dump = java.lang.NullPointerException
	at com.ibm.ws.opentracing.OpentracingUserFeatureAccessService.getTracerInstance(OpentracingUserFeatureAccessService.java:66)
	at com.ibm.ws.opentracing.OpentracingTracerManager.createTracer(OpentracingTracerManager.java:98)
	at com.ibm.ws.opentracing.OpentracingTracerManager.ensureTracer(OpentracingTracerManager.java:73)
	at com.ibm.ws.opentracing.OpentracingTracerManager.getTracer(OpentracingTracerManager.java:132)
	at com.ibm.ws.opentracing.OpentracingContainerFilter.filter(OpentracingContainerFilter.java:74)
	at org.apache.cxf.jaxrs.utils.JAXRSUtils.runContainerRequestFilters(JAXRSUtils.java:1761)
	at org.apache.cxf.jaxrs.interceptor.JAXRSInInterceptor.processRequest(JAXRSInInterceptor.java:275)
	at org.apache.cxf.jaxrs.interceptor.JAXRSInInterceptor.handleMessage(JAXRSInInterceptor.java:90)
	at org.apache.cxf.phase.PhaseInterceptorChain.doIntercept(PhaseInterceptorChain.java:308)
	at org.apache.cxf.transport.ChainInitiationObserver.onMessage(ChainInitiationObserver.java:124)
	at org.apache.cxf.transport.http.AbstractHTTPDestination.invoke(AbstractHTTPDestination.java:279)
	at com.ibm.ws.jaxrs20.endpoint.AbstractJaxRsWebEndpoint.invoke(AbstractJaxRsWebEndpoint.java:134)
	at com.ibm.websphere.jaxrs.server.IBMRestServlet.handleRequest(IBMRestServlet.java:146)
	at com.ibm.websphere.jaxrs.server.IBMRestServlet.doPost(IBMRestServlet.java:104)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:707)
	at com.ibm.websphere.jaxrs.server.IBMRestServlet.service(IBMRestServlet.java:96)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.service(ServletWrapper.java:1255)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:743)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:440)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.invokeTarget(WebAppFilterChain.java:147)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.doFilter(WebAppFilterChain.java:89)
	at com.recres.web.server.BootstrapFilter.doFilter(BootstrapFilter.java:33)
	at com.ibm.ws.webcontainer.filter.FilterInstanceWrapper.doFilter(FilterInstanceWrapper.java:201)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.doFilter(WebAppFilterChain.java:86)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.doFilter(WebAppFilterManager.java:995)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1124)
	at com.ibm.ws.webcontainer.webapp.WebApp.handleRequest(WebApp.java:4962)
	at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.handleRequest(DynamicVirtualHost.java:314)
	at com.ibm.ws.webcontainer.WebContainer.handleRequest(WebContainer.java:995)
	at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.run(DynamicVirtualHost.java:279)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink$TaskWrapper.run(HttpDispatcherLink.java:1009)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.wrapHandlerAndExecute(HttpDispatcherLink.java:412)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.ready(HttpDispatcherLink.java:371)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleDiscrimination(HttpInboundLink.java:530)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleNewRequest(HttpInboundLink.java:464)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.processRequest(HttpInboundLink.java:329)
	at com.ibm.ws.http.channel.internal.inbound.HttpICLReadCallback.complete(HttpICLReadCallback.java:70)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.requestComplete(WorkQueueManager.java:501)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.attemptIO(WorkQueueManager.java:571)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.workerRun(WorkQueueManager.java:926)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager$Worker.run(WorkQueueManager.java:1015)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
```

I should say that fixing the underlying issue in my service fixes the OpenTracing NPE too, but that is no the point ... OpenTracing should be more robust...

**Comment by NottyCode** [comment]
This was fixed a month ago by Issue #2928. I know the title doesn't necessarily sound like it, but the code changes prevent the null pointer exception you experienced. So this should be fixed in the next release.

**Comment by hrstoyanov** [comment]
Thanks! .. But if so, when can we get this into an official downloadable build? I only see 18.0.0.1 at the web site?
